### DISCUSSION TITLE
Understanding an error

### DISCUSSION THREAD

**Discussion by MichaelBMiner** [cantools/cantools#460] [Q&A]
Hello,

I am having trouble understanding and error I am seeing. It is `Signed integer value 53134 out of range.` My code and DBC are very simple. My message and signals are:
```
BO_ 2566540286 PGN_64075_Aftertreatment_2_SCR_Dosing_System_Information_5: 8  EV
 SG_ Aftertreatment_2_Diesel_Exhaust_Fluid_Doser_2_Pressure_Extended_Range : 0|16@1- (0.1,-100) [-100|6325.5] "kPa" EVCC
 SG_ Aftertreatment_2_Diesel_Exhaust_Fluid_Doser_3_Pressure_Extended_Range : 16|16@1- (0.1,-100) [-100|6325.5] "kPa" EVCC
```

My code is very simple. I am trying to use cantools and my DBC to generate random data to flood my DUT with. 

```
import cantools
import can
import random

can_bus = can.interface.Bus('vcan0', bustype='socketcan')
db = cantools.database.load_file("DefaultDBC.dbc")


while True:
    for msg in db.messages:
        ext_can_id = msg.frame_id
        data_dict = {}
        data = None
        if msg.is_multiplexed():
            pass
        else:
            for sig in msg.signals:
                if "." in str(sig.minimum)  or "." in str(sig.maximum):
                    sig_data = round(random.uniform(sig.minimum, sig.maximum), 2)
                else:
                    sig_data = random.randrange(sig.minimum, sig.maximum)
                data_dict[sig.name] = sig_data
            try:            
                data = db.encode_message(ext_can_id, data_dict)
            except Exception as err:
                #print(data_dict, err)
                print(sig.name, sig_data, int(sig_data), err)
                
            message = can.Message(arbitration_id=ext_can_id, data=data)
            can_bus.send(message)
```

The error I am seeing is as follows:
`Aftertreatment_1_Diesel_Exhaust_Fluid_Doser_3_Pressure_Extended_Range 5213.4 5213 Signed integer value 53134 out of range.`

This confuses me as the values, float and int and within the appropriate range of my DBC signals range. What am I missing here?

**Comment by juleq** [comment]
Hi.
- The 16bit signed integer range is -32768 to 32767.
- Physical value = raw value * factor + offset.
- So raw = (phy-offset)/factor ~= (5300+100)/0.1 = 54000.
- 54000 > 32767.

I assume you might have wanted the factor to be 10?

Hint: My brain does suck at remembering binary. The way I remember the order this is supposed to be calculated is the following: Factor and offset both carry the unit of the physical value. The raw value is unitless. If you can recall that, you will be able to reconstruct the correct sequence those values are applied to get from one to the other. I have to do that every single time :).
